test/filtrex_test.exs

```diff
@@ -178,8 +178,9 @@ defmodule FiltrexTest do
                   "comparator" => "equals",
                   "value" => "2016-03-26",
                   "type" => "date"
-                }
-              ]
+                },
+              ],
+              "sub_filters" => []
```

Comment:
Have you actually seen the recursion working properly? It looks like the code is right, but it might be worth it to throw in an example where we have a sub-filter that has to be dumped. (I'll do a final review to make sure we didn't miss anything, but everything in the new commits look good.)